### PR TITLE
ADCM-1515 add /api/v1/component/ path to API

### PR DESCRIPTION
**Pull request by grey-kristy**

### CODE REVIEW

File: python/api/component/serializers.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+class ComponentObjectUrlField(serializers.HyperlinkedIdentityField):
+    def get_url(self, obj, view_name, request, format):
+        kwargs = {'component_id': obj.id}
+        return reverse(view_name, kwargs=kwargs, request=request, format=format)
+
+
+class ComponentSerializer(serializers.Serializer):
+    id = serializers.IntegerField(read_only=True)
+    cluster_id = serializers.IntegerField(required=True)
```
Comment (a-alferov) [inline python/api/component/serializers.py:34]: Please add the `service_id` field. Also, please bring the data for all endpoints to one view. `/api/v1/cluster/1/service/1/component/`, `/api/v1/service/1/component/` and `/api/v1/component/`, and also for detailed display.

File: python/api/api_views.py [hunk 2]
```diff
@@ -100,8 +100,9 @@ def get_api_url_kwargs(obj, request):
         if 'cluster' in request.path:
             kwargs['cluster_id'] = obj.cluster.id
     elif obj_type == 'component':
-        kwargs['service_id'] = obj.service.id
-        kwargs['cluster_id'] = obj.cluster.id
+        if 'cluster' in request.path:
```
Comment (Maunty) [inline python/api/api_views.py:103]:
```suggestion
        elif obj_type == 'component' and 'cluster' in request.path:
```
@pisohaka 
Check plz sonarcube setup.
In report there is a code smell warning but check is green, can we fail it ?

File: python/api/component/serializers.py (hunk 1 above)
Comment (grey-kristy) [reply python/api/component/serializers.py]: all fixed

**Review by a-alferov — APPROVED** [review]

**Review by a-alferov — CHANGES_REQUESTED** [review]
Please remove the filtering

File: python/api/component/views.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+from api.api_views import PageView, check_obj, DetailViewRO
+from cm.models import ServiceComponent, ClusterObject, Cluster
+from . import serializers
+
+
+class ComponentListView(PageView):
+    queryset = ServiceComponent.objects.all()
+    serializer_class = serializers.ComponentSerializer
+    serializer_class_ui = serializers.ComponentUISerializer
+    filterset_fields = ('cluster_id', )
```
[the remark this answers is not in the document]
Comment (a-alferov) [reply python/api/component/views.py]: Canceled

**Review by a-alferov — APPROVED** [review]

**Pull request merged**